packages/test/src/test-workflows.ts

```diff
@@ -1544,4 +1571,120 @@ test('continue-as-new-same-workflow', async (t) => {
   }
 });
 
+test('patched:not-replay', async (t) => {
+  const { logs, script } = t.context;
+  {
+    const req = await activate(t, makeStartWorkflow(script));
+    compareCompletion(
+      t,
+      req,
+      makeSuccess([
+        makeSetChangeMarker('my-change-id', false),
+        makeStartTimerCommand({ timerId: '0', startToFireTimeout: msToTs(100) }),
+      ])
+    );
+  }
+  {
+    const req = await activate(t, makeFireTimer('0'));
+    compareCompletion(
+      t,
+      req,
+      makeSuccess([makeSetChangeMarker('my-change-id', false), makeCompleteWorkflowExecution()])
+    );
+  }
+  t.deepEqual(logs, [['has change'], ['has change 2']]);
+});
+
+test('patched:replay-no-marker', async (t) => {
+  const { logs, script } = t.context;
+  {
+    const req = await activate(t, makeStartWorkflow(script, undefined, undefined, true));
+    compareCompletion(
+      t,
+      req,
+      makeSuccess([
+        makeSetChangeMarker('my-change-id', false),
```

Comment:
We probably shouldn't send this command out since we didn't use the change branch.

Same goes for the second marker.